### ISSUE TITLE
Create demo course for xblock-poll

### ISSUE THREAD

**Issue by bradenmacdonald** [open-craft/demo-courses#1]
We need a demo course for [xblock-poll](https://github.com/open-craft/xblock-poll).

Requirements:
- The course should at least cover the same material as the README which can be seen at https://github.com/open-craft/xblock-poll
- In other words, the new course should explain what "xblock-poll" is, explain how to set it up, and then show examples of what it can do.
- [The examples](https://github.com/open-craft/xblock-poll#poll-examples) in the new demo course should be working poll/survey components and not just screenshots.

Notes:
- You may copy the examples from the README screenshots, or create brand new examples.
- You can still include the screenshots if you'd like, below the working examples, because they can show more realistic vote breakdowns. This course will be reset fairly often and whenever the course has been reset, the first few learners to use the course will not see very interesting poll/survey results since the number of respondents will be very low. That's why the screenshots help.)
-  To add a poll/survey example to a course in "Studio", click on this button:
  ![screen shot 2016-05-21 at 12 28 42 am](https://cloud.githubusercontent.com/assets/945577/15447131/18f2c952-1eeb-11e6-8554-c4871663a1ad.png)
  - Then click "Poll" or "Survey". Once it appears, click Edit" to edit it.

**Comment by bradenmacdonald** [comment]
_(Moving follow-up discussion from email thread...)_

I had set up a course template on our http://studio.dogwood.sandbox.opencraft.com/, but then that sandbox was deleted. 

@antoviaque on May 25:

> Rose Regina, to unblock you, you can use http://courses.opencraft.com instead. There is also the demo course there, and I've created a course with sample poll & survey blocks, so you have an example of that too.

@antoviaque on June 1:

> I don't think I have seen any activity on this over the last week. Did I miss anything?

Rose Regina then said on June 2:

> I thought I had replied at the end of last week, but found my response sitting as a draft. 
> 
> Basically, I think the delete was probably actually for the better. I had started writing put material and playing around a little bit, though not getting particularly. I think what I came up with this time is a lot better. I believe that you should now be able to see what I had added. While most of it is from the documentation on github, I added a few things that I thought could use clarification and reordered a few bits. 
> 
> There are maybe a few points that could use additional clarification and screenshots, but I think this is probably good for people who are already familiar with Xblocks (though being not quite there myself, it's a bit hard to gauge). 

I reviewed the course so far:

The course looks good! I think you've done a nice job with what I consider a fairly challenging task. Thanks :)
- On this page http://courses.opencraft.com/courses/course-v1:Test+TST+now/courseware/12bba176574d46d9b42ac0b4068049f1/d81a4831f87e4a56ab86ec8d9fbf1980/ I was hoping that more of the examples could be converted to live examples. It is currently confusing that so many of them are screenshots, since it looks like they are real interactive polls but clicking on them doesn't work. I think we need to either indicate more clearly that they are screenshots (e.g. put a border around them) or, preferably, convert them to actual example polls. In the task description I had mentioned including screenshots but mostly to show the "vote breakdown" part, not the "choose an option to vote for" part.

Minor notes:
- A lot of the images appear slightly distorted - the aspect ratio is wrong. In the HTML of your "Text" areas, most of the images specify a width and height; if you remove those and/or just use 'style="width: auto; max-width: 100%"' it will fix that.
- The screenshot images are mostly links to GitHub; the potential issue with that is that if we delete the images on GitHub, they will disappear from the course. It's better to upload the images using the "Content > Files & Uploads" menu in Studio, then linking to the simple URL it gives (e.g. '/static/Screen_Shot_2016-06-01_at_2.15.10_PM.png'). I wouldn't expect you to know that though, so no worries about that.
- http://courses.opencraft.com/courses/course-v1:Test+TST+now/courseware/78cccba3aa8b4c22bd96035f4ac566d0/cf1b4508354148fbb0cf1849da1a5015/ Typo: "Yes, compleatly"
- http://courses.opencraft.com/courses/course-v1:Test+TST+now/courseware/12bba176574d46d9b42ac0b4068049f1/97299891cdc143da8a9030d1ea66770b/ Sentence fragment in the feedback of the poll: "If you are confused about markdown, check out."

**Comment by roseregina** [comment]
agreed that the screen shots are a little confusing. I have started going through and converting them to actual polls. thank you for the info about images! I wasn't sure of the best approach and that makes a lot more sense than what I was doing. I'm also going through and changing where the images are stored and linked to as I correct the polls. 

I am not seeing the "yes, compleatly" typo on that screen. could you tell me the name of the unit it is in? 

the broken syntax in the markdown link is fixed now. 

also, 
1 . I saw that a translation section had been added for the github page, so I went ahead and added that to the course as well. 

2 . I want to point out (though you have probably already noticed) that the naming scheme has a high potential for confusion given that the XBlock is named polls, but there is also a polls component (but not the only component). I don't have a good suggestion, but it seemed like it might be early enough to change something and make it clearer for end users. 

3 . with adding the poll and survey under advanced settings menu, it is depicted as: 
<code> [

```
 "poll",

 "survey"
```

] </code>
(apologies for the formatting on that but I'm having a hard tiem fixing it and I think you'll get the idea)
would it also work as?: 
<code>[ "poll", "survey" ] </code> 
and if not, should there be a reminder that the line breaks are important or is it safe to assume that the course designer will know and remember that? 

4 . I left out the screenshot of the Survey ith Markdown and Images because the total came to more than %100 +/- rounding and that does not seem possible from my understanding. if that just an odd fluke, or is there a way to let people select more than one answer per questions? and if so, how? because that is something that I think should definitely be included in the tutorial.

**Comment by roseregina** [comment]
I finished the edits to the course, aside from potentially adding borders around the screenshots of poll results since that didn't seem to be an easy option within the course and manually doing it would have been quite time intensive, but I think that things should be a lot more clear now. 

there was one image that appeared not to load correctly in the studio, but it looks fine in the course view, so I'm going to go ahead and ignore whatever was going on with that ( "poll edit screenshot" in Creating Polls and Surveys > Making Polls and Surveys > Using the Studio).

**Comment by bradenmacdonald** [comment]
@roseregina 

> I finished the edits to the course

Yay! Thanks :) It looks good.

> I am not seeing the "yes, compleatly" typo on that screen. could you tell me the name of the unit it is in?

Sorry, I thought I had linked directly to it, but apparently not. Here's where it is:
![screen shot 2016-06-06 at 3 11 52 pm](https://cloud.githubusercontent.com/assets/945577/15839609/0e44a2d0-2bf9-11e6-8e95-e3b390a35306.png)
1. Awesome - thanks!
2. What do you mean? Do you mean that it's confusing that "polls" is the name of the overall package which includes the Poll XBlock and the Survey XBlock?  The terms "XBlock" and "component" mean the same thing in Studio. (Technically a "component" means either an XBlock or an XModule, which is an older precursor to XBlocks.)
3. Good question. Yes, the spaces and newlines are ignored so either way works.
4. Re " I left out the screenshot of the Survey ith Markdown and Images because the total came to more than %100 +/- rounding": Hmm, that sounds like a bug. Can you show the screenshot here? Just drag a screenshot onto one of these comment fields to include it in the GitHub comment.

> there was one image that appeared not to load correctly in the studio, but it looks fine in the course view, so I'm going to go ahead and ignore whatever was going on with that ( "poll edit screenshot" in Creating Polls and Surveys > Making Polls and Surveys > Using the Studio).

Weird. It seems to be working for me in both Studio and the LMS, so let's not worry about that for now.

> aside from potentially adding borders around the screenshots of poll results since that didn't seem to be an easy option within the course

You can add `style="border: 4px solid grey;"` to each image in the HTML edit mode, but that's kind of a hassle, and I don't think it's necessary anymore after your other changes.

**Comment by roseregina** [comment]
> What do you mean? Do you mean that it's confusing that "polls" is the name of the overall package which includes the Poll XBlock and the Survey XBlock? The terms "XBlock" and "component" mean the same thing in Studio. (Technically a "component" means either an XBlock or an XModule, which is an older precursor to XBlocks.)
> yes, basically, that in trying to write clearly about the two (polls as overall package and polls as "Poll XBlock), I see a potential for confusion, but like I said, I don't have a better suggestion. you have a lot more experience with what people actually end up finding confusing and what is fine, I just noticed it is the kind of thing in slightly different contexts I'm used to people not having the easiest time with. 
1. 
   here is the link to the image in github: https://github.com/open-craft/xblock-poll/blob/master/doc_img/survey_mixed_result.png
   first line is fine, but the following two are impossible. 

also, I went ahead and fixed the typo in the first poll.

**Comment by bradenmacdonald** [comment]
Good catch with those nonsensical results. Thanks. I agree we should omit that screenshot.

**Comment by samuelallan72** [comment]
@bradenmacdonald looks like this was completed, and safe to close the issue?

**Comment by bradenmacdonald** [comment]
@samuelallan72 Wow, old issue. Sure.
